crates/mpt/src/hp.rs

```diff
@@ -165,6 +165,59 @@ pub(crate) fn encoded_path_strip_prefix<'a>(
     Some(&nibs[i..])
 }
 
+/// Encodes nibbles into the standard hex-prefix format directly into the bump arena.
+#[inline]
+pub(crate) fn to_encoded_path_with_bump<'a>(
+    bump: &'a bumpalo::Bump,
+    nibs: &[u8],
+    is_leaf: bool,
+) -> &'a [u8] {
+    let is_odd = nibs.len() % 2 != 0;
+    // Max path is 64 nibs (32 bytes) + 1 prefix byte = 33 bytes.
+    let mut encoded = bumpalo::collections::Vec::with_capacity_in(33, bump);
+
+    let mut prefix = if is_leaf { 0x20 } else { 0x00 };
+    if is_odd {
+        prefix |= 0x10;
+        encoded.push(prefix | nibs[0]);
+        for i in (1..nibs.len()).step_by(2) {
+            encoded.push((nibs[i] << 4) | nibs[i + 1]);
+        }
+    } else {
+        encoded.push(prefix);
+        for i in (0..nibs.len()).step_by(2) {
+            encoded.push((nibs[i] << 4) | nibs[i + 1]);
+        }
+    }
+
+    encoded.into_bump_slice()
+}
+
+/// Encodes nibbles into the standard hex-prefix format.
+#[cfg(feature = "host")]
```

Comment:
nit: we use `#[cfg(target_os = "zkvm")]` in some places. Maybe it could be used here as well for consistency?

Reply:
That works too but I think using a feature is cleaner. I feel like `#[cfg(target_os = "zkvm")]` is more appropriate for cases where you want to customize a thing depending on target OS.